dashboard/app/helpers/users_helper.rb

```diff
@@ -21,8 +21,16 @@ def check_and_apply_oauth_takeover(user)
       oauth_token = session[ACCT_TAKEOVER_OAUTH_TOKEN]
       clear_takeover_session_variables
 
-      # TODO: validate that we're not destroying an active account?
       existing_account = User.find_by_credential(type: provider, id: uid)
+      if existing_account.has_activity?
+        # We don't want to destroy an account with progress.
+        # In theory this should not happen, so we log a Honeybadger error and return.
+        Honeybadger.notify(
+          error_class: 'Oauth takeover called for user with progress',
+          error_message: "Attempted to take over account with id #{existing_account.id}, which has activity"
```

Comment:
I _think_ this Honeybadger notification will capture information about the logged-in user automatically, so between that and your including the `existing_account.id` here we should be able to tell which two accounts are involved.  Is that correct?

Reply:
Yes, that's my expectation.